src/command_modules/azure-cli-container/azure/cli/command_modules/container/custom.py

```diff
@@ -188,13 +190,27 @@ def create_container(cmd,
                             ip_address=cgroup_ip_address,
                             image_registry_credentials=image_registry_credentials,
                             volumes=volumes or None,
+                            network_profile=_get_network_profile(network_profile, ip_address, dns_name_label),
                             diagnostics=diagnostics,
                             tags=tags)
 
     container_group_client = cf_container_groups(cmd.cli_ctx)
     return sdk_no_wait(no_wait, container_group_client.create_or_update, resource_group_name, name, cgroup)
 
 
+def _get_network_profile(network_profile, ip_address, dns_name_label):
+    if network_profile and ip_address:
+        raise CLIError('Can not use "--network-profile" with IP address type "Public".')
+    elif network_profile and dns_name_label:
+        raise CLIError('Can not use "--network-profile" with "--dns_name_label".')
```

Comment:
This should be `--dns-name-label`.  In general this kind of validation should be done in a validator or deferred to the service unless as a workaround when the service error message is inadequate.